src/python/impactx/dashboard/Input/generalFunctions.py

```diff
@@ -0,0 +1,273 @@
+import inspect
+import os
+import re
+import subprocess
+import webbrowser
+
+from ..trame_setup import setup_server
+
+# -----------------------------------------------------------------------------
+# Trame setup
+# -----------------------------------------------------------------------------
+
+server, state, ctrl = setup_server()
+
+# -----------------------------------------------------------------------------
+# Code
+# -----------------------------------------------------------------------------
+ANSI_RED = "\033[91m"
+ANSI_RESET = "\033[0m"
+
+
+class generalFunctions:
+
+    @staticmethod
+    def documentation(section_name):
+        """
+        Opens a tab to the specified section link in the documentation.
+        :param section_name (str): The name of the documentation section to open.
+        """
+
+        if section_name == "LatticeElements":
+            url = "https://impactx.readthedocs.io/en/latest/usage/python.html#lattice-elements"
+        elif section_name == "BeamDistributions":
+            url = "https://impactx.readthedocs.io/en/latest/usage/python.html#initial-beam-distributions"
+        elif section_name == "pythonParameters":
+            url = "https://impactx.readthedocs.io/en/latest/usage/python.html#general"
+        else:
+            raise ValueError(f"Invalid section name: {section_name}")
+
+        if "WSL_DISTRO_NAME" in os.environ:
+            subprocess.run(["explorer.exe", url])
+        else:
+            webbrowser.open_new_tab(url)
+
+    # -----------------------------------------------------------------------------
+    # Validation functions
+    # -----------------------------------------------------------------------------
+
+    @staticmethod
+    def determine_input_type(value):
+        """
+        Determines the type of the input value.
+        :param value: The input value whose type needs to be determined.
+        :return: A tuple containing the value converted to its determined type and the type itself.
+        """
+
+        try:
+            return int(value), int
+        except ValueError:
+            try:
+                return float(value), float
+            except ValueError:
+                return value, str
+
+    @staticmethod
+    def validate_against(input_value, value_type):
+        """
+        Returns an error message if the input value type does not match the desired type.
+        :param input_value: The value to validate.
+        :param value_type: The desired type ('int', 'float', 'str').
+        :return: A list of error messages. An empty list if there are no errors.
+        """
+
+        if value_type == "int":
+            if input_value is None:
+                return ["Must be an integer"]
+            try:
+                value = int(input_value)
+                if value < -10000000000:
+                    return ["Must be positive"]
```

Comment:
This looks confusing to me. Should this read
```suggestion
                if value < 0:
                    return ["Must be positive"]
```
?

Reply:
Yes, we need to fix this. I remember intentionally changing '<0" to '<-10000000000' before. I've seen that some parameters can be negative values (ie. k parameter in drift element), and having the GUI not allow negative values was an issue. In a future PR, we can focus on setting boundaries for specific parameters?

Reply:
Sounds good! Yes, let's remove this for now, strict positive values depend a lot on the exact parameter and quite a few allow negative values, too.